File: zig/isogram/Iteration1/isogram.zig
pub fn isIsogram(str: []const u8) bool {

    var characterFlags: u32 = 0;

    for (str) |character| {

        var charIndex: u5 = 0;
        
        if (character >= 'a' and character <= 'z') {
            charIndex = @intCast(character - 'a');
        }
        else if (character >= 'A' and character <= 'Z') {
            charIndex = @intCast(character - 'A');
        }
        else {
            continue;
        }

        const mask = @as(u32, 1) << charIndex;

        if (characterFlags & mask != 0) return false;
        characterFlags |= mask;
    }

    return true;
}

File: c/isogram/Iteration1/isogram.c

#include <stdbool.h>
#include <stddef.h>

#include "isogram.h"

bool is_isogram(const char phrase[])
{
    if (phrase == NULL) return false;
    if (*phrase == '\0') return true;
    
    unsigned int characterCounts[26] = { 0 };
    
    for (const char* c = phrase; *c != '\0'; c++)
    {
        char character = *c;

        // Lowercase letters...
        if (character >= 97 && character <= 122)
        {
            int charIndex = character - 97;
            characterCounts[charIndex]++;
        }

        // Uppercase letters...
        if (character >= 65 && character <= 90)
        {
            int charIndex = character - 65;
            characterCounts[charIndex]++;
        }
    }
    
    for (int i = 0; i < 26; i++)
    {
        if (characterCounts[i] > 1) return false;
    }
    
    return true;
}